### PR TITLE
Feature/rh test caf

### PR DESCRIPTION
**Pull request by rhowell42**
I changed the basic variables and cuts to accommade the new CAF reco structure change. These have the same functionality as before, where the desired track or shower is grabbed that has the appropriate trackscore. This request depends on other pull requests in SBNSoftware/sbnanaobj and SBNSoftware/sbncode.

https://github.com/SBNSoftware/sbnanaobj/tree/feature/rh_testCAF
https://github.com/SBNSoftware/sbncode/tree/feature/rh_allPFOs

### CODE REVIEW

File: sbnana/SBNAna/Cuts/NumuCuts.cxx [hunk 1]
```diff
@@ -7,11 +7,11 @@ namespace ana{
 
   const Cut kNumuBasicQual([](const caf::SRSliceProxy* slc)
 			   { // 
-			     bool hastrk = ( slc->reco.ntrk > 0 );
+			     bool hastrk = ( slc->reco.npfp > 0 );
 			     if (!hastrk) return hastrk;
 
 			     unsigned int muIdx = (unsigned int)kPrimMuonIdx(slc);
- 			     double len = slc->reco.trk[muIdx].len;
+ 			     double len = slc->reco.pfp[muIdx].shw.len;
```
Comment (etyley) [inline sbnana/SBNAna/Cuts/NumuCuts.cxx:14]: Lots of this file has `trk` converted to `shw`

**Review by etyley — CHANGES_REQUESTED** [review]
Looks like some of the changes were mangled in the transition

**Comment by etyley** [comment]
We also probably need to have a collective discussion about the best way to merge these sort of breaking changes in an upcoming meeting, otherwise develop will not work on any of the production that we just made.

**Comment by cjbackhouse** [comment]
Discussion is scheduled at SBN AI tomorrow. The idea I proposed before was to keep the trk and shw branches in parallel in the StandardRecord structure for one release. Then you can try to access whichever (trk/shw or pfp) you think exists in your files.

File: sbnana/SBNAna/Cuts/NumuCuts.cxx (hunk 1 above)
Comment (rhowell42) [reply sbnana/SBNAna/Cuts/NumuCuts.cxx:14]: Ah right. I'll go through and fix them. Thanks

**Comment by rhowell42** [comment]
So you want me to restore the reco->trk/shw branches for that pull request?

**Comment by rhowell42** [comment]
I fixed and pushed the fixes for the badly copied lines above, and also fixed a nue cut

File: sbnana/SBNAna/Cuts/NueCuts.cxx [hunk 2]
```diff
@@ -134,8 +134,9 @@ namespace ana{
   const Cut kNueAllTrackDazzleMuonCut(
     [](const caf::SRSliceProxy* slc)
     {
-      for (auto const& trk : slc->reco.trk)
+      for (auto const& pfp : slc->reco.pfp)
       {
+      const auto& trk = pfp.trk;
```
Comment (etyley) [inline sbnana/SBNAna/Cuts/NueCuts.cxx:139]: I think this is missing a check on the track score if we want to be completely backwards compatible

File: sbnana/SBNAna/Cuts/NueCuts.cxx [hunk 3]
```diff
@@ -146,8 +147,9 @@ namespace ana{
   const Cut kNueAllTrackDazzleMuonScoreCut(
     [](const caf::SRSliceProxy* slc)
     {
-      for (auto const& trk : slc->reco.trk)
+      for (auto const& pfp : slc->reco.pfp)
```
Comment (etyley) [inline sbnana/SBNAna/Cuts/NueCuts.cxx:150]: I think this is missing a check on the track score if we want to be completely backwards compatible

File: sbnana/SBNAna/Cuts/NumuCuts.cxx [hunk 4]
```diff
@@ -7,11 +7,11 @@ namespace ana{
 
   const Cut kNumuBasicQual([](const caf::SRSliceProxy* slc)
 			   { // 
-			     bool hastrk = ( slc->reco.ntrk > 0 );
+			     bool hastrk = ( slc->reco.npfp > 0 );
 			     if (!hastrk) return hastrk;
 
 			     unsigned int muIdx = (unsigned int)kPrimMuonIdx(slc);
- 			     double len = slc->reco.trk[muIdx].len;
+ 			     double len = slc->reco.pfp[muIdx].trk.len;
```
Comment (etyley) [inline sbnana/SBNAna/Cuts/NumuCuts.cxx:14]: I guess that `muIdx` can now feasibly be `-1` so should be checked

File: sbnana/SBNAna/Cuts/NumuCuts.cxx [hunk 5]
```diff
@@ -31,11 +31,11 @@ namespace ana{
   
   const Cut kNumuTrkLen([](const caf::SRSliceProxy* slc)
 			{ // TO DO: Prim pandora trk len primary_track.length > fConfig.TrackLength
-			  bool hastrk = ( slc->reco.ntrk > 0 );
+			  bool hastrk = ( slc->reco.npfp > 0 );
 			  if (!hastrk) return hastrk;
 			  
 			  unsigned int muIdx = (unsigned int)kPrimMuonIdx(slc);
-			  double len = slc->reco.trk[muIdx].len;
+			  double len = slc->reco.pfp[muIdx].trk.len;
```
Comment (etyley) [inline sbnana/SBNAna/Cuts/NumuCuts.cxx:38]: I guess that `muIdx` can now feasibly be `-1` so should be checked

File: sbnana/SBNAna/Vars/NueVars.cxx [hunk 6]
```diff
@@ -144,7 +144,8 @@ const Var kRecoShower_trackWidth(
 const Var kRecoShowers_EnergyCut(
     [](const caf::SRSliceProxy* slc) -> unsigned {
       unsigned int counter(0);
-      for (auto const& shw : slc->reco.shw) {
+      for (auto const& pfp : slc->reco.pfp) {
```
Comment (etyley) [inline sbnana/SBNAna/Vars/NueVars.cxx:147]: I think this is missing a check on the track score if we want to be completely backwards compatible

**Review by etyley — COMMENTED** [review]
I think there are a couple places where the behaviour is slightly changed, but it otherwise looks good to me

File: sbnana/SBNAna/Cuts/NumuCuts.cxx (hunk 5 above)
Comment (rhowell42) [reply sbnana/SBNAna/Cuts/NumuCuts.cxx:38]: I can put in a check for muIdx > 0 like exists in other parts of this repo, which should fix any issues

Comment (etyley) [reply sbnana/SBNAna/Cuts/NumuCuts.cxx:38]: Any updates on implementing this?

**Review by etyley — APPROVED** [review]

**Pull request merged**